cavalier_contours_ffi/src/lib.rs

```diff
@@ -247,6 +292,216 @@ fn boolean_op_from_u32(i: u32) -> Option<BooleanOp> {
     }
 }
 
+/// FFI representation of SelfIntersectsInclude enum
+pub const CAVC_SELF_INTERSECTS_INCLUDE_ALL: u32 = 0;
+pub const CAVC_SELF_INTERSECTS_INCLUDE_LOCAL: u32 = 1;
+pub const CAVC_SELF_INTERSECTS_INCLUDE_GLOBAL: u32 = 2;
+
+/// FFI representation of [PlineSelfIntersectOptions].
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct cavc_pline_self_intersect_o {
+    pub pline_aabb_index: *const cavc_aabbindex,
+    pub pos_equal_eps: f64,
+    pub include: u32,
+}
+
+impl cavc_pline_self_intersect_o {
+    /// Convert FFI self intersection options type to internal type.
+    ///
+    /// # Safety
+    ///
+    /// `pline_aabb_index` field must be null or a valid pointer to a [cavc_aabbindex].
+    pub unsafe fn to_internal(&self) -> PlineSelfIntersectOptions<f64> {
+        let pline_aabb_index = unsafe { self.pline_aabb_index.as_ref().map(|w| &w.0) };
+        let include_value = match self.include {
+            0 => SelfIntersectsInclude::All,
+            1 => SelfIntersectsInclude::Local,
+            2 => SelfIntersectsInclude::Global,
+            _ => panic!(
+                "Invalid include option in cavc_pline_self_intersect_options_o::to_internal: {}",
+                self.include
+            ),
+        };
```

Comment:
I don't think we want to just panic here as that will crash the caller going through the C FFI, probably should bubble up as error return in the C FFI API.